source/github-helpers/parse-compare-url.test.ts

```diff
@@ -0,0 +1,45 @@
+import {expect, test} from 'vitest';
+
+import parseCompareUrl from './parse-compare-url.js';
+
+const base = {
+	branch: 'main',
+	repo: {
+		nameWithOwner: 'john/recipes',
+	},
+};
+
+test('parseCompareUrl', () => {
+	expect(parseCompareUrl('/john/recipes/compare')).toBe(undefined);
+	expect(parseCompareUrl('/john/recipes/compare/main')).toBe(undefined);
```

Comment:
Behavior change"

Without information of the other branch, there's no comparison, it's always empty. We need to "handle" these URLs by just returning early and returning undefined/false